src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -255,45 +256,43 @@ public void drive(double xSpeed, double ySpeed, double rotRate, boolean fieldRel
     SwerveDriveKinematics.desaturateWheelSpeeds(
         swerveModuleStates, DriveConstants.kMaxSpeedMetersPerSecond);
-    m_frontLeft.setDesiredState(swerveModuleStates[0]);
-    m_frontRight.setDesiredState(swerveModuleStates[1]);
-    m_rearLeft.setDesiredState(swerveModuleStates[2]);
-    m_rearRight.setDesiredState(swerveModuleStates[3]);
-
-    swerveModuleDesiredStatePublisher.set(swerveModuleStates);
-
+    swerveModuleStates[0] = frontLeft.setDesiredStateAndGetDesiredState(swerveModuleStates[0]);
+    swerveModuleStates[1] = frontRight.setDesiredStateAndGetDesiredState(swerveModuleStates[1]);
+    swerveModuleStates[2] = rearLeft.setDesiredStateAndGetDesiredState(swerveModuleStates[2]);
+    swerveModuleStates[3] = rearRight.setDesiredStateAndGetDesiredState(swerveModuleStates[3]);
   }
 
   /**
    * Sets the wheels into an X formation to prevent movement.
    */
   public void setX() {
-    m_frontLeft.setDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(45)));
-    m_frontRight.setDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(-45)));
-    m_rearLeft.setDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(-45)));
-    m_rearRight.setDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(45)));
+    swerveModuleStates[0] = frontLeft
+        .setDesiredStateAndGetDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(45)));
+    swerveModuleStates[1] = frontRight
+        .setDesiredStateAndGetDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(-45)));
+    swerveModuleStates[2] = rearLeft
+        .setDesiredStateAndGetDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(-45)));
+    swerveModuleStates[3] = rearRight
+        .setDesiredStateAndGetDesiredState(new SwerveModuleState(0, Rotation2d.fromDegrees(45)));
   }
 
   public ChassisSpeeds getRobotRelativeSpeeds() {
-    return DriveConstants.kDriveKinematics.toChassisSpeeds(m_frontLeft.getState(), m_frontRight.getState(),
-        m_rearLeft.getState(), m_rearRight.getState());
+    return DriveConstants.kDriveKinematics.toChassisSpeeds(frontLeft.getState(), frontRight.getState(),
+        rearLeft.getState(), rearRight.getState());
   }
 
   public void setRobotRelativeSpeeds(ChassisSpeeds speeds) {
     speeds = ChassisSpeeds.discretize(speeds, .02);
-    var swerveModuleStates = DriveConstants.kDriveKinematics.toSwerveModuleStates(speeds);
+    swerveModuleStates = DriveConstants.kDriveKinematics.toSwerveModuleStates(speeds);
     SwerveDriveKinematics.desaturateWheelSpeeds(
         swerveModuleStates, DriveConstants.kMaxSpeedMetersPerSecond);
-    m_frontLeft.setDesiredState(swerveModuleStates[0]);
-    m_frontRight.setDesiredState(swerveModuleStates[1]);
-    m_rearLeft.setDesiredState(swerveModuleStates[2]);
-    m_rearRight.setDesiredState(swerveModuleStates[3]);
-
-    swerveModuleDesiredStatePublisher.set(swerveModuleStates);
-
+    frontLeft.setDesiredStateAndGetDesiredState(swerveModuleStates[0]);
```

Comment:
is there a reason we aren't updating the swerve module states as we're setting them here?